static/usr/lib/systemd/system/var-lib-snapd-save.mount

```diff
@@ -0,0 +1,8 @@
+[Unit]
+ConditionPathIsMountPoint=/run/mnt/ubuntu-save
+
+[Mount]
+What=/run/mnt/ubuntu-save
+Where=/var/lib/snapd/save
```

Comment:
does this doesn't need any requirement keywords because systemd does the right thing in terms of dependencies checking having var/lib mounted before trying this?

Reply:
This is the effective properties of the unit (only dependencies):

```
Requires="dev-mapper-ubuntu\\x2dsave\\x2dbafd47f3\\x2da887\\x2d4061\\x2d997f\\x2dcf8a073b38bb.device" "run-mnt-ubuntu\\x2dsave.mount" system.slice var-lib-snapd.mount -.mount
RequiredBy=local-fs.target
Conflicts=umount.target
Before=umount.target local-fs.target
After="blockdev@dev-mapper-ubuntu\\x2dsave\\x2dbafd47f3\\x2da887\\x2d4061\\x2d997f\\x2dcf8a073b38bb.target" "run-mnt-ubuntu\\x2dsave.mount" "dev-mapper-ubuntu\\x2dsave\\x2dbafd47f3\\x2da887\\x2d4061\\x2d997f\\x2dcf8a073b38bb.device" systemd-journald.socket system.slice var-lib-snapd.mount -.mount local-fs-pre.target
StopPropagatedFrom="dev-mapper-ubuntu\\x2dsave\\x2dbafd47f3\\x2da887\\x2d4061\\x2d997f\\x2dcf8a073b38bb.device"
RequiresMountsFor=/run/mnt/ubuntu-save /var/lib/snapd
```

So I am not sure where it is documented that `RequiresMountsFor` is automatic.

Reply:
*If a mount unit is beneath another mount unit in the file system hierarchy, both a requirement dependency and an ordering dependency between both units are created automatically.* AFAIU systemd will automatically detect any filesystem tree dependencies and order/depend on those

Reply:
It is state in the "implicit dependencies" of the `systemd.mount` manual page.